File: 746.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        vector<int> results(cost.size(),0); 
        results[0] = cost[0]; 
        results[1] = cost[1];  
        for (int i = 2; i < results.size(); i++){
            results[i] = cost[i] + min(results[i-2], results[i-1]); 
        }
        
        return min(results[cost.size()-1], results[cost.size()-2]); 
    }
};

File: 746.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        elem_1 = cost[1]    # curr_pos - 1 
        elem_2 = cost[0]    # curr_pos - 2 
        
        cost_size = len(cost)
        
        for i in range(2, cost_size): 
            aux = elem_1
            elem_1 = min(elem_1, elem_2) + cost[i]
            elem_2 = aux 
            
        return min(elem_1, elem_2)
        
        
